frontend/src/lib/components/LemonTextArea/LemonTextArea.tsx

```diff
@@ -63,13 +68,118 @@ interface LemonTextMarkdownProps {
+        e.preventDefault()
+        e.stopPropagation()
+        setDrag(false)
+        if (e.dataTransfer?.files && e.dataTransfer.files.length > 0) {
+            try {
+                setIsUploading(true)
+                const formData = new FormData()
+                formData.append('image', e.dataTransfer.files[0])
+                const media = await api.media.upload(formData)
+                onChange(value + `\n\n![${media.name}](${media.image_location})`)
+                posthog.capture('markdown image uploaded', { name: media.name })
+            } catch (error) {
+                const errorDetail = (error as any).detail || 'unknown error'
+                posthog.capture('markdown image upload failed', { error: errorDetail })
+                lemonToast.error(`Error uploading image: ${errorDetail}`)
+            } finally {
+                setIsUploading(false)
+            }
+            e.dataTransfer.clearData()
+            dragCounter = 0
+        }
+    }
+
+    useEffect(() => {
+        const div = dropRef.current
+        if (!div || !objectStorageAvailable) {
+            return
+        } else {
+            div.addEventListener('dragenter', handleDragIn)
+            div.addEventListener('dragleave', handleDragOut)
+            div.addEventListener('dragover', handleDrag)
+            div.addEventListener('drop', handleDrop)
+            return () => {
+                div?.removeEventListener('dragenter', handleDragIn)
+                div?.removeEventListener('dragleave', handleDragOut)
+                div?.removeEventListener('dragover', handleDrag)
+                div?.removeEventListener('drop', handleDrop)
+            }
+        }
+    }, [value, objectStorageAvailable])
+
     return (
         <Tabs>
             <Tabs.TabPane tab="Write" key="write-card" destroyInactiveTabPane={true}>
-                <LemonTextArea {...editAreaProps} autoFocus value={value} onChange={(newValue) => onChange(newValue)} />
-                <div className="text-muted inline-flex items-center space-x-1">
-                    <IconMarkdown className={'text-2xl'} />
-                    <span>Markdown formatting support</span>
+                <div
+                    ref={dropRef}
+                    className={clsx('LemonTextMarkdown flex flex-col p-2 space-y-1 rounded', drag && 'FileDropTarget')}
+                >
+                    <LemonTextArea {...editAreaProps} autoFocus value={value} onChange={onChange} />
+                    <div className="text-muted inline-flex items-center space-x-1">
+                        <IconMarkdown className={'text-2xl'} />
+                        <span>Markdown formatting support</span>
+                    </div>
+                    {objectStorageAvailable ? (
+                        <div className="text-muted inline-flex items-center space-x-1">
+                            <IconUploadFile className={'text-2xl mr-1'} />
```

Comment:
I suppose `IconMarkdown` should have `mr-1` too

Reply:
The width of the Markdown icon made the text look off to me. So `mr` on the two other icons to get the text closer to lining up

<img width="498" alt="Screenshot 2022-10-13 at 18 09 24" src="https://user-
images.githubusercontent.com/984817/195662081-a65fe30e-e540-40e8-8ea6-da03c53a95e6.png">